### DISCUSSION TITLE
Support for BAT tokens and IPSF?

### DISCUSSION THREAD

**Discussion by uwla** [qutebrowser/qutebrowser#6349] [Ideas] [answered]
Hello, thanks for the fantastic browser.

Is there any plans to add support to BAT (Basic Attention Takes) like the Brave Browser? If not, why? There a couple of vim and Linux users who are content creators and used to use qutebrowser but migrated to Brave so they can get rewards for their content (such as luke smith and brodie robertson).

What about the  IPFS (Interplanetary file system) ? This is a protocol to transfer files in peer-to-peer and Brave supports that. Few people access IPFS websites but it will likely became more widespread as advances in Web 3.0 are made. (I, for example, have Brave only to access content through IPFS. For the rest I use qutebrowser)

Adding support to BAT and IPFS could make qutebrowser more popular. I guess IPFS would be less important than BAT right now because it is something relative new and there are more people interested in earning BAT.

**Comment by The-Compiler** [comment, marked as answer]
> Is there any plans to add support to BAT (Basic Attention Takes) like the Brave Browser? If not, why?

Nope, not in qutebrowser's core at least. For a variety of reasons:

- It's questionable how much benefit there really is to content creators (also see below). Honestly the main party benefiting from it is probably Brave with their US$35M ICO sale.
- Many content creators might not even be interested in that (yet Brave injects their BAT logo for all of my commits, at least last time I checked).
- It's unclear if this is even feasible (or possible at all) to implement.
- Code collecting and selling your data just doesn't belong in qutebrowser, period.

If someone tells me what's needed to implement this, it might be possible to expose those things in a future extension API (#30), and if someone develops an extension for it, it's not like I'm going to block it or anything (as long as it's not actively malicious at least). But my interest for working on this is around zero.

> There a couple of vim and Linux users who are content creators and used to use qutebrowser but migrated to Brave so they can get rewards for their content (such as luke smith and brodie robertson).

Do you have a source on that? Why would getting rewards for their contents require them using Brave for their own day-to-day browsing? This makes no sense to me.

Also, just as a reality check: Brodie Robertson, with some 15K subscribers, has a [recent video about Brave](https://www.youtube.com/watch?v=F4tnsng3Yv0) where he [mentions](https://youtu.be/F4tnsng3Yv0?t=160) he gets about 4.5 BAT per month for using Brave, and around 38 BAT per month in donations. That [seems to translate](https://coinmarketcap.com/currencies/basic-attention-token/) to:

- About USD 6 per month earned by using Brave
- About USD 48 per month in donations (though only about USD 8 at the beginning of this year, so it really depends on how the price develops)

It's far from nothing of course, but (especially with the first value of those two) is hardly a good reason for someone to insist on not using qutebrowser.

> What about the IPFS (Interplanetary file system) ? This is a protocol to transfer files in peer-to-peer and Brave supports that. Few people access IPFS websites but it will likely became more widespread as advances in Web 3.0 are made. (I, for example, have Brave only to access content through IPFS. For the rest I use qutebrowser)

That's already listed in #30, also see #1314. This is more likely to happen given that it's clearer what qutebrowser would actually need to do to add support (and there already is [a project](https://github.com/pinnaculum/galacteek) based on Qt supporting it) - but again, I'll probably not be working on it myself.

> Adding support to BAT and IPFS could make qutebrowser more popular.

FWIW that's not necessarily a goal. The audience of qutebrowser is relatively niche - it's not for everyone, but that's entirely okay. I'd rather have happy users than many users - and for the forseeable future, I'll be busy working on things affecting a much larger amount of users than this would.

**Reply by uwla** [reply]
I see your point. BAT is probably mainly benefiting Brave, but also a few users (I wish I earned $50 a month with it like Brodie). Even for a normal user (not content creator) it's possible to get around 2-4 BAT per month (which is enough to pay for a droplet on DO or to get some coffe) just for browsing around. I do agree with the other of your points. About an API for that, If I get to know that kind of stuff, I'll get in touch 

About IPFS, great! Thanks for providing a link to the QT project supporting it.

Lastly, my point here was just suggest 2 features that I **guess** could be good (but it's just my opinion anyway). I don't mind if these features are not implemented. I like qutebrowser and know it has its focus. Best regards
